Agent/SessionReplay/SessionReplayManager.swift

```diff
@@ -28,16 +30,18 @@ public class SessionReplayManager: NSObject {
 
     public func start() {
         sessionReplay.start()
+        isFirstChunck = true
         guard !isRunning() else {
             NRLOG_WARNING("Session replay harvest timer attempting to start while already running.")
             return
         }
 
-        NRLOG_DEBUG("Session replay harvest timer starting with a period of \(harvestPeriod) ms")
-
-        self.harvestTimer = Timer(timeInterval: TimeInterval(self.harvestPeriod), target: self, selector: #selector(self.harvest), userInfo: nil, repeats: true)
-
-        RunLoop.current.add(self.harvestTimer!, forMode: .default)
+        NRLOG_DEBUG("Session replay harvest timer starting with a period of \(harvestPeriod) s")
+        DispatchQueue.main.async { [self] in
```

Comment:
This is pretty interesting, is it only required to use the main thread because of use of Timer()? Should we consider using 
DispatchQueue.global(qos: .background)  {
Timer.scheduledTimer(withTimeInerval: TimeInterval(self.harvestPeriod)) {

}

}